homeassistant/components/unifi/sensor.py

```diff
@@ -192,6 +199,137 @@ def async_device_state_value_fn(hub: UnifiHub, device: Device) -> str:
+
+    return item is not None
+
+
+def make_wan_latency_sensors(
+    wan: Literal["WAN", "WAN2"],
+) -> tuple[UnifiSensorEntityDescription, ...]:
+    """Create WAN latency sensors from WAN monitor data."""
+
+    return (
+        UnifiSensorEntityDescription[Devices, Device](
+            key="Microsoft WAN latency",
+            entity_category=EntityCategory.DIAGNOSTIC,
+            native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+            state_class=SensorStateClass.MEASUREMENT,
+            device_class=SensorDeviceClass.DURATION,
+            entity_registry_enabled_default=False,
+            api_handler_fn=lambda api: api.devices,
+            available_fn=async_device_available_fn,
+            device_info_fn=async_device_device_info_fn,
+            name_fn=lambda _: "Microsoft WAN latency",
+            object_fn=lambda api, obj_id: api.devices[obj_id],
+            supported_fn=partial(async_device_latency_supported_fn, wan, "microsoft"),
+            unique_id_fn=lambda hub, obj_id: f"microsoft_wan_latency-{obj_id}",
+            value_fn=partial(async_client_wan_monitor_latency, wan, "microsoft"),
+        ),
+        UnifiSensorEntityDescription[Devices, Device](
+            key="Google WAN latency",
+            entity_category=EntityCategory.DIAGNOSTIC,
+            native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+            state_class=SensorStateClass.MEASUREMENT,
+            device_class=SensorDeviceClass.DURATION,
+            entity_registry_enabled_default=False,
+            api_handler_fn=lambda api: api.devices,
+            available_fn=async_device_available_fn,
+            device_info_fn=async_device_device_info_fn,
+            name_fn=lambda _: "Google WAN latency",
+            object_fn=lambda api, obj_id: api.devices[obj_id],
+            supported_fn=partial(async_device_latency_supported_fn, wan, "google"),
+            unique_id_fn=lambda hub, obj_id: f"google_wan_latency-{obj_id}",
+            value_fn=partial(async_client_wan_monitor_latency, wan, "google"),
+        ),
+        UnifiSensorEntityDescription[Devices, Device](
+            key="Cloudflare WAN latency",
+            entity_category=EntityCategory.DIAGNOSTIC,
+            native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+            state_class=SensorStateClass.MEASUREMENT,
+            device_class=SensorDeviceClass.DURATION,
+            entity_registry_enabled_default=False,
+            api_handler_fn=lambda api: api.devices,
+            available_fn=async_device_available_fn,
+            device_info_fn=async_device_device_info_fn,
+            name_fn=lambda _: "Cloudflare WAN latency",
+            object_fn=lambda api, obj_id: api.devices[obj_id],
+            supported_fn=partial(async_device_latency_supported_fn, wan, "1.1.1.1"),
+            unique_id_fn=lambda hub, obj_id: f"cloudflare_wan_latency-{obj_id}",
+            value_fn=partial(async_client_wan_monitor_latency, wan, "1.1.1.1"),
+        ),
+    )
```

Comment:
Last thing is probably to combine these into one method that takes the source as input